packages/next/server/render.tsx

```diff
@@ -1490,6 +1530,7 @@ function renderToReadableStream(
         if (!done) {
           const s = typeof value === 'string' ? value : decoder.decode(value)
           if (shellCompleted) {
+            res.write(getFlushPrefix())
```

Comment:
We can’t have this here, it should only be inserted once right after the shell is completed.

Reply:
No, we need to write it before flushing writes from React. We can throttle it (with `setTimeout` or `queueMicrotask`) but we still need it for more than just the first flush.

Reply:
I meant it should be put in the `else` branch of this if-else condition here. Here `res.write` will be called multiple times whenever there is new data ready to write to the stream.